static/app/views/profiling/landing/chart.tsx

```diff
@@ -0,0 +1,184 @@
+import {useMemo} from 'react';
+import {browserHistory, withRouter, WithRouterProps} from 'react-router';
+import {useTheme} from '@emotion/react';
+import {Location} from 'history';
+
+import ChartZoom from 'sentry/components/charts/chartZoom';
+import OptionSelector from 'sentry/components/charts/optionSelector';
+import ScatterChart from 'sentry/components/charts/scatterChart';
+import {
+  ChartContainer,
+  ChartControls,
+  InlineContainer,
+} from 'sentry/components/charts/styles';
+import TransitionChart from 'sentry/components/charts/transitionChart';
+import TransparentLoadingMask from 'sentry/components/charts/transparentLoadingMask';
+import {getSeriesSelection} from 'sentry/components/charts/utils';
+import {Panel} from 'sentry/components/panels';
+import Placeholder from 'sentry/components/placeholder';
+import {t} from 'sentry/locale';
+import {SelectValue} from 'sentry/types';
+import {EChartClickHandler, SeriesDataUnit} from 'sentry/types/echarts';
+import {Trace} from 'sentry/types/profiling/trace';
+import {defined} from 'sentry/utils';
+import {axisLabelFormatter, tooltipFormatter} from 'sentry/utils/discover/charts';
+import getDynamicText from 'sentry/utils/getDynamicText';
+import {decodeScalar} from 'sentry/utils/queryString';
+
+interface Props extends WithRouterProps {
+  location: Location;
+  traces: Trace[];
+  end?: string;
+  start?: string;
+  statsPeriod?: string | null;
+  utc?: string;
+}
+
+function _ProfileScatterChart({
+  router,
+  location,
+  traces,
+  start,
+  end,
+  statsPeriod,
+  utc,
+}: Props) {
+  const theme = useTheme();
+
+  const colorEncoding = useMemo(() => getColorEncodingFromLocation(location), [location]);
+
+  const series = useMemo(() => {
```

Comment:
I like adding return types here (variable declaration or useMemo return, either works), but it's something that helps keep tsc errors local and avoid some back and forth in case you make a mistake you see it here right away instead of where the series is passed to other components